### PR TITLE
Propose to download the task result

### PR DESCRIPTION
**Pull request by cedric25**
+ unzip and display result.txt content

Two new questions at the end of `iaddp run`:

Capture d’écran 2024-12-09 à 17 56 37

### CODE REVIEW

File: cli/src/cmd/run.js [hunk 1]
```diff
@@ -186,12 +188,49 @@ export async function runInDebug({
...
+    spinner.stop();
+    process.exit(1);
+  }
+
+  spinner.start('Downloading result...');
+
+  const taskResult = await iexec.task.fetchResults(taskId);
+  const resultBuffer = await taskResult.arrayBuffer();
+
+  const outputFolder = CACHE_DIR + '/result';
```
Comment (cedric25) [inline cli/src/cmd/run.js:214]: Choosing to store downloaded zip into /cache/result.
It's open for discussion.

**Comment by PierreJeanjacquot** [comment]
@cedric25 some code already exists to display the output properly, here is my proposal to factorize the code and offer a similar experience while running `iapp test` and `iapp run` https://github.com/iExecBlockchainComputing/idapp/pull/30

File: cli/src/utils/extractZipToFolder.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+async function fileExists(path) {
+  try {
+    await fs.access(path);
+    return true;
+  } catch {
+    return false;
+  }
+}
```
Comment (PierreJeanjacquot) [inline cli/src/utils/extractZipToFolder.js:37]: already exposed by `fileExists.js` module (fixed in #30)

File: cli/src/cmd/run.js [hunk 3]
```diff
@@ -199,12 +201,49 @@ export async function runInDebug({
...
+  await extractZipToFolder(resultBuffer, outputFolder);
```
[the remark this answers is not in the document]
Comment (cedric25) [reply cli/src/cmd/run.js:228]: https://github.com/iExecBlockchainComputing/iapp/pull/30/commits/b044147b93750841fae65115b1224af0c116c663

**Comment by cedric25** [comment]
Closing in favour of https://github.com/iExecBlockchainComputing/iapp/pull/30

Thanks for improving this code :)

**Pull request closed without merging**